File: jsfring-webservice/src/main/java/fr/pinguet62/jsfring/webservice/converter/PageMapper.java
package fr.pinguet62.jsfring.webservice.converter;

import fr.pinguet62.jsfring.webservice.dto.PageDto;
import lombok.experimental.UtilityClass;
import org.springframework.data.domain.Page;

import java.util.List;
import java.util.function.Function;

import static java.util.stream.Collectors.toList;

@UtilityClass
public class PageMapper {

    public <Entity, Dto> PageDto<Dto> toDto(Page<Entity> entity, Function<Entity, Dto> elementMapper) {
        if (entity == null)
            return null;
        List<Dto> dtos = entity.stream().map(elementMapper).collect(toList());
        return new PageDto<>(dtos, entity.getTotalElements());
    }

}

File: jsfring-webservice/src/main/kotlin/fr/pinguet62/jsfring/webservice/converter/PageMapper.kt
package fr.pinguet62.jsfring.webservice.converter

import fr.pinguet62.jsfring.webservice.dto.PageDto
import org.springframework.data.domain.Page
import java.util.stream.Collectors.toList

fun <Entity, Dto> toDto(entities: Page<Entity>, elementMapper: (Entity) -> Dto): PageDto<Dto> {
//    if (entities == null)
//        return null
    var dtos: List<Dto> = entities.stream().map(elementMapper).collect(toList()) // TODO Kotlin
    return PageDto(dtos, entities.totalElements)
}
